internal/mode/static/state/graph/nginxproxy_test.go

```diff
@@ -356,3 +362,137 @@ func TestValidateNginxProxy(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateRewriteClientIP(t *testing.T) {
+	tests := []struct {
+		np             *ngfAPI.NginxProxy
+		validator      *validationfakes.FakeGenericValidator
+		name           string
+		errorString    string
+		expectErrCount int
+	}{
+		{
+			name:      "valid rewriteClientIP",
+			validator: createValidValidator(),
+			np: &ngfAPI.NginxProxy{
+				Spec: ngfAPI.NginxProxySpec{
+					RewriteClientIP: &ngfAPI.RewriteClientIP{
+						SetIPRecursively: helpers.GetPointer(true),
+						TrustedAddresses: []ngfAPI.TrustedAddress{"2001:db8:a0b:12f0::1/32", "10.56.32.11/32"},
+						Mode:             helpers.GetPointer(ngfAPI.RewriteClientIPModeProxyProtocol),
+					},
+				},
+			},
+			expectErrCount: 0,
+		},
+		{
+			name:      "invalid CIDR in trustedAddresses",
+			validator: createInvalidValidator(),
+			np: &ngfAPI.NginxProxy{
+				Spec: ngfAPI.NginxProxySpec{
+					RewriteClientIP: &ngfAPI.RewriteClientIP{
+						SetIPRecursively: helpers.GetPointer(true),
+						TrustedAddresses: []ngfAPI.TrustedAddress{"2001:db8::/129", "10.0.0.1"},
+						Mode:             helpers.GetPointer(ngfAPI.RewriteClientIPModeProxyProtocol),
+					},
+				},
+			},
+			expectErrCount: 1,
+			errorString: "spec.rewriteClientIP.trustedAddresses.2001:db8::/129: " +
+				"Invalid value: \"2001:db8::/129\": must be a valid IP address or CIDR range",
+		},
+		{
+			name:      "invalid IP and CIDR in trustedAddresses",
+			validator: createInvalidValidator(),
+			np: &ngfAPI.NginxProxy{
+				Spec: ngfAPI.NginxProxySpec{
+					RewriteClientIP: &ngfAPI.RewriteClientIP{
+						SetIPRecursively: helpers.GetPointer(true),
+						TrustedAddresses: []ngfAPI.TrustedAddress{"2001:db8::1/48", "256.100.50.25"},
+						Mode:             helpers.GetPointer(ngfAPI.RewriteClientIPModeProxyProtocol),
+					},
+				},
+			},
+			expectErrCount: 1,
+			errorString: "spec.rewriteClientIP.trustedAddresses.256.100.50.25: " +
```

Comment:
Shouldn't we get two errors in this case -- one for each invalid address?

Reply:
"2001:db8::1/48", "256.100.50.25" --> only address that's invalid is `256.100.50.25`

Reply:
I see. Can you change the test description to be more explicit? Maybe: "invalid IP and valid CIDR in trustedAddresses"